mapie/time_series_regression.py

```diff
@@ -0,0 +1,312 @@
+            Input data.
+
+        y : ArrayLike of shape (n_samples_test,)
+            Input labels.
+
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+
+        Raises
+        ------
+        ValueError
+            If the lenght of y is greater than the lenght of the training set.
+        """
+        X = cast(NDArray, X)
+        y = cast(NDArray, y)
+        if len(X) > len(self.conformity_scores_):
+            raise ValueError(
+                "You try to update more residuals than there are!"
+            )
+        new_conformity_scores_ = self._relative_conformity_scores(X, y)
+        self.conformity_scores_ = np.roll(
+            self.conformity_scores_, -len(new_conformity_scores_)
+        )
+        self.conformity_scores_[-len(new_conformity_scores_):] = (
+            new_conformity_scores_
+        )
+        return self
+
+    def _beta_optimize(
+        self,
+        alpha: Union[float, NDArray],
+        upper_bounds: NDArray,
+        lower_bounds: NDArray,
+    ) -> NDArray:
+        """
+        ``_beta_optimize`` offers to minimize the width of the PIs, for a given
+        difference of quantiles.
+
+        Parameters
+        ----------
+        alpha: Union[float, NDArray]
+            The quantiles to compute.
+        upper_bounds: NDArray
+            The array of upper values.
+        lower_bounds: NDArray
+            The array of lower values.
+
+        Returns
+        -------
+        NDArray
+            Array of betas minimizing the differences
+            ``(1-alpa+beta)-quantile - beta-quantile``.
+
+        Raises
+        ------
+        ValueError
+            If lower and upper bounds arrays don't have the same shape.
+        """
+        if lower_bounds.shape != upper_bounds.shape:
+            raise ValueError(
+                "Lower and upper bounds arrays should have the same shape."
+            )
+        alpha = cast(NDArray, alpha)
+        betas_0 = np.full(
+            shape=(len(alpha), len(lower_bounds)),
+            fill_value=np.nan,
+            dtype=float,
+        )
+
+        for ind_alpha, _alpha in enumerate(alpha):
+            betas = np.linspace(
+                _alpha / (len(lower_bounds) + 1),
+                _alpha,
+                num=len(lower_bounds),
+                endpoint=True,
+            )
+            one_alpha_beta = np_nanquantile(
+                upper_bounds,
+                1 - _alpha + betas,
+                axis=1,
+                method="higher",
+            )  # type: ignore
+            beta = np_nanquantile(
+                lower_bounds,
+                betas,
+                axis=1,
+                method="lower",
+            )  # type: ignore
+            betas_0[ind_alpha, :] = betas[
+                np.argmin(one_alpha_beta - beta, axis=0)
+            ]
+
+        return betas_0
+
+    def _pred_multi(self, X: ArrayLike) -> NDArray:
```

Comment:
Move this method directly to `MapieRegressor` and used it also in the base class ?

Reply:
Done